src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -371,7 +373,7 @@ private async Task ExecuteAsync(HttpContext context, ConnectionDelegate connecti
                 }
 
                 // Mark the connection as active
-                connection.Status = HttpConnectionStatus.Active;
+                connection.ChangeState(HttpConnectionStatus.Inactive, HttpConnectionStatus.Active);
```

Comment:
Should we just continue on if the state transition fails? Should we assert the transitions are successful?

Reply:
I think it's fine to let it continue

Reply:
What if you're in the HttpConnectionStatus.Disposed state?

Reply:
It can't be because they both use the same lock and there is a check for Disposed state after acquiring the lock in this case.